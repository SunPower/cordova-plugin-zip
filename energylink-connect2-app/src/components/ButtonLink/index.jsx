import React from 'react'
import { useHistory } from 'react-router-dom'
import './ButtonLink.scss'

export const ButtonLink = ({ title = 'Link Title', path }) => {
  const history = useHistory()

  return (
    <div className="button-link mt-10 mb-10" onClick={() => history.push(path)}>
      <div className="route-name">
        <span>{title}</span>
      </div>
      <div className="arrow-right pt-2">
        <span className="sp-chevron-right has-text-primary" />
      </div>
    </div>
  )
}
