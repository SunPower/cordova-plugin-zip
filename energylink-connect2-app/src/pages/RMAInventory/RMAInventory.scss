@import '~@sunpower/theme-dark/styles/variables';

.rma-add-devices {
  display: flex;
  flex-direction: column;
  min-height: 100%;

  .rma-form {
    flex-grow: 1;

    [type=number] {
      color: $black;
    }
  }

  .rma-buttons {
    display: flex;
    justify-content: center;
  }

  .checkbox {
    grid-area: checkbox;
    -webkit-appearance: none;
    background-color: $black;
    border: 1px solid $silver-sand2;
    padding: 10px;
    border-radius: 3px;
    display: inline-block;
    position: relative;

    &:checked:after {
      content: '\2713';
      font-size: 20px;
      position: absolute;
      top: -4px;
      left: 1px;
      color: #99a1a7;
    }
  }
}
