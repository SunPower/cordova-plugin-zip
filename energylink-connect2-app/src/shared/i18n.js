import { useMemo } from 'react'
import { useSelector } from 'react-redux'

import * as languages from './locales'
import * as localizedComponents from './localizedComponents'

const getTranslate = translations => (key, ...params) => {
  let value = translations[key] || key
  if (params) {
    params.forEach((p, idx) => {
      value = value.replace(`{${idx}}`, p)
    })
  }
  return value
}

export const useI18n = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language
  return useMemo(() => getTranslate(languages[locale]), [locale])
}

export const useI18nComponent = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language || {}

  return useMemo(
    () => componentName => {
      const path = `${locale}_${componentName}`
      const Component = localizedComponents[path] || {}
      if (typeof Component !== 'function') {
        throw new Error(`Component not found: ${path}`)
      }
      return Component()
    },
    [locale]
  )
}
