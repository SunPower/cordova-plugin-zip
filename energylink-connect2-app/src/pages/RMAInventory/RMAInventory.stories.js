import React from 'react'
import { storiesOf } from '@storybook/react'
import { configureStore } from 'state/store'
import { Provider } from 'react-redux'

import RMAInventory from './index'

const inventoryStore = {
  inventory: {
    bom: [
      { item: 'AC_MODULES', value: '0' },
      { item: 'DC_MODULES', value: '0' },
      { item: 'STRING_INVERTERS', value: '0' },
      { item: 'EXTERNAL_METERS', value: '0' },
      { item: 'ESS', value: '0' }
    ],
    rma: {
      other: true
    }
  }
}

storiesOf('RMA - Inventory', module).add('Simple', () => {
  const { store } = configureStore(inventoryStore)

  return (
    <div className="full-min-height">
      <Provider store={store}>
        <RMAInventory />
      </Provider>
    </div>
  )
})
