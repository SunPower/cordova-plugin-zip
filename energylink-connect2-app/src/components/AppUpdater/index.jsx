import React from 'react'

import { useI18n } from 'shared/i18n'

function AppUpdater({ onUpdate }) {
  const t = useI18n()

  return (
    <section className="is-flex tile is-vertical level file">
      <h1 className="has-text-centered">{t('UPDATE_HEADER')}</h1>

      <button
        className="button is-primary mt-30 is-uppercase"
        onClick={onUpdate}
      >
        {t('BTN_UPDATE')}
      </button>
    </section>
  )
}

export default AppUpdater
