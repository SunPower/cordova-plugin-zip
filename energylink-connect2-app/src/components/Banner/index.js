import React from 'react'
import './banner.scss'
import { either, isDebug } from 'shared/utils'

const Banner = ({ flavor = '' }) =>
  either(
    isDebug,
    <div className="is-capitalized development-banner">
      {flavor.split('-').pop()}
    </div>
  )

export default Banner
