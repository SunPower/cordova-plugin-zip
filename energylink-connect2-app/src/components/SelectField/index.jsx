import React from 'react'
import clsx from 'clsx'
import Select, { components } from 'react-select'
import { either } from 'shared/utils'

import './SelectField.scss'

const IndicatorSeparator = ({ selectProps: { required } }) =>
  either(required, <span className="ml-20 pt-5 has-text-danger">*</span>)

const Option = props => {
  return (
    <components.Option {...props}>
      <div className="is-flex file level">
        <p className="is-size-6">{props.label}</p>
      </div>
    </components.Option>
  )
}

const selectComponents = { Option, IndicatorSeparator }

function SelectField(props) {
  const {
    notFoundText,
    onSelect,
    className,
    autoFocus,
    options,
    defaultValue,
    placeholder,
    isSearchable,
    value,
    disabled,
    required
  } = props

  const classes = clsx('field', className)

  const defaults = {
    cacheOptions: true,
    placeholder,
    onChange: onSelect,
    className: classes,
    classNamePrefix: 'search',
    autoFocus,
    components: selectComponents,
    options,
    defaultValue,
    isSearchable,
    value,
    isDisabled: disabled,
    hideSelectedOptions: true,
    noOptionsMessage: () => notFoundText,
    theme: theme => ({
      ...theme,
      colors: {
        ...theme.colors,
        primary50: '#0a1421',
        primary25: '#0a1421',
        primary: 'transparent',
        neutral0: '#2f465b',
        neutral90: 'white',
        neutral70: 'white',
        neutral80: 'white',
        neutral5: '#35404e'
      }
    }),
    required
  }

  return <Select {...defaults} />
}

export default SelectField
