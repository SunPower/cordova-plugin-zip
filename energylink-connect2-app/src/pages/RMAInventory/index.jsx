import React from 'react'
import { find, prop, propEq } from 'ramda'
import { useDispatch, useSelector } from 'react-redux'
import { useHistory } from 'react-router-dom'
import paths from 'routes/paths'
import { useI18n } from 'shared/i18n'
import {
  UPDATE_INVENTORY,
  UPDATE_OTHER_INVENTORY
} from 'state/actions/inventory'
import { START_DISCOVERY_INIT } from 'state/actions/pvs'
import { discoveryTypes } from 'state/reducers/devices'
import { SHOW_MODAL } from 'state/actions/modal'
import SelectField from 'components/SelectField'

import './RMAInventory.scss'

const RMAInventory = () => {
  const t = useI18n()
  const dispatch = useDispatch()
  const { bom, rma } = useSelector(state => state.inventory)
  const miValue = find(propEq('item', 'AC_MODULES'), bom)
  const storageValue = find(propEq('item', 'ESS'), bom)
  const history = useHistory()

  const essOptions = [
    { value: '0', label: t('NONE') },
    { value: '13kWh', label: t('13KWH_1INV') },
    { value: '26kWh (1 inverter)', label: t('26KWH_1INV') },
    { value: '26kWh (2 inverters)', label: t('26KWH_2INV') }
  ]

  const miOptions = [...Array(99).keys()].map(number => ({
    label: number,
    value: number
  }))

  const handleCheckbox = e => {
    e.preventDefault()
    dispatch(UPDATE_OTHER_INVENTORY(e.target.checked))
  }

  const handleChange = name => option => {
    dispatch(
      UPDATE_INVENTORY({
        name: name,
        value: option.value
      })
    )
  }

  const handleContinue = () => {
    if (prop('other', rma)) {
      dispatch(
        START_DISCOVERY_INIT({
          Device: 'allplusmime',
          type: discoveryTypes.LEGACY
        })
      )
      history.push(paths.PROTECTED.LEGACY_DISCOVERY.path)
    } else if (parseInt(miValue.value) > 0) {
      history.push(paths.PROTECTED.SCAN_LABELS.path)
    } else if (parseInt(storageValue.value) !== 0) {
      history.push(paths.PROTECTED.STORAGE_PREDISCOVERY.path)
    } else {
      dispatch(
        SHOW_MODAL({
          title: t('FIX_ERROR_TO_PROCEED'),
          body: t('RMA_SELECT_DEVICE_ERROR'),
          dismissable: true,
          withButtons: true
        })
      )
    }
  }

  const handleCancel = () => {
    history.push(paths.PROTECTED.RMA_DEVICES.path)
  }

  return (
    <div className="rma-add-devices pr-10 pl-10">
      <div className="rma-form">
        <p className="is-uppercase has-text-centered has-text-weight-bold">
          {t('RMA_ADD_DEVICES_TITLE')}
        </p>

        <div className="collapsible mt-20">
          <div className="collapsible-header">
            <div className="collapsible-title">
              <div className="has-text-weight-bold" htmlFor="MI">
                {t('MICROINVERTERS')}
              </div>
            </div>
          </div>

          <div className="field mt-10">
            <SelectField
              isSearchable={false}
              onSelect={handleChange('AC_MODULES')}
              defaultValue={find(propEq('value', miValue.value), miOptions)}
              options={miOptions}
            />
          </div>
        </div>

        <div className="collapsible mt-10">
          <div className="collapsible-header">
            <div className="collapsible-title">
              <label className="has-text-weight-bold" htmlFor="other">
                {t('SUNVAULT_STORAGE')}
              </label>
            </div>
          </div>
          <div className="field mt-10">
            <SelectField
              isSearchable={false}
              onSelect={handleChange('ESS')}
              value={find(propEq('value', storageValue.value), essOptions)}
              options={essOptions}
            />
          </div>
        </div>

        <div className="collapsible mt-10">
          <div className="collapsible-header">
            <div className="collapsible-title">
              <input
                type="checkbox"
                id="other"
                name="other"
                onChange={handleCheckbox}
                className="mr-10 checkbox-dark"
                defaultChecked={prop('other', rma)}
              />
              <label className="has-text-weight-bold" htmlFor="other">
                {t('OTHER_DEVICES')}
              </label>
            </div>
          </div>
        </div>
      </div>

      <div>
        <p className="has-text-centered mt-20 mb-20">
          {t('RMA_ADD_DEVICES_TIP')}
        </p>

        <div className="rma-buttons">
          <button
            className="button half-button-padding is-secondary is-uppercase trigger-scan mr-10"
            onClick={handleCancel}
          >
            {t('CANCEL')}
          </button>
          <button
            className="button half-button-padding is-primary is-uppercase trigger-scan"
            onClick={handleContinue}
          >
            {t('CONTINUE')}
          </button>
        </div>
      </div>
    </div>
  )
}

export default RMAInventory
