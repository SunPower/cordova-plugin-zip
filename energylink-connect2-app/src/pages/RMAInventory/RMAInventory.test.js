import React from 'react'
import * as i18n from 'shared/i18n'
import * as SelectField from 'components/SelectField'
import RMAInventory from '.'

describe('RMAInventory', () => {
  let state = {
    inventory: {
      bom: [
        { item: 'AC_MODULES', value: '0' },
        { item: 'DC_MODULES', value: '0' },
        { item: 'STRING_INVERTERS', value: '0' },
        { item: 'EXTERNAL_METERS', value: '0' },
        { item: 'ESS', value: '0' }
      ],
      rma: {
        other: true
      }
    }
  }

  beforeEach(() => {
    jest
      .spyOn(i18n, 'useI18n')
      .mockImplementation(() => (key, ...params) =>
        `${key.toUpperCase()} ${params.join('_')}`.trim()
      )

    jest.spyOn(SelectField, 'default').mockImplementation(() => <div />)
  })

  test('renders correctly', () => {
    const { component } = mountWithProvider(<RMAInventory />)(state)
    expect(component).toMatchSnapshot()
  })
})
