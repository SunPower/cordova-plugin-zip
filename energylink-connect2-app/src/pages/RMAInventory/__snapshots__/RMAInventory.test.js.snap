// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RMAInventory renders correctly 1`] = `
<BrowserRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <RMAInventory>
        <div
          className="rma-add-devices full-height pr-20 pl-20"
        >
          <div
            className="rma-form"
          >
            <p
              className="is-uppercase has-text-centered has-text-weight-bold"
            >
              RMA_ADD_DEVICES_TITLE
            </p>
            <div
              className="collapsible mt-20"
            >
              <div
                className="collapsible-header"
              >
                <div
                  className="collapsible-title"
                >
                  <div
                    className="has-text-weight-bold"
                    htmlFor="MI"
                  >
                    MICROINVERTERS
                  </div>
                </div>
              </div>
              <div
                className="field  mb-20 mt-20"
              >
                <mockConstructor
                  isSearchable={false}
                  onSelect={[Function]}
                  options={
                    Array [
                      Object {
                        "label": 0,
                        "value": 0,
                      },
                      Object {
                        "label": 1,
                        "value": 1,
                      },
                      Object {
                        "label": 2,
                        "value": 2,
                      },
                      Object {
                        "label": 3,
                        "value": 3,
                      },
                      Object {
                        "label": 4,
                        "value": 4,
                      },
                      Object {
                        "label": 5,
                        "value": 5,
                      },
                      Object {
                        "label": 6,
                        "value": 6,
                      },
                      Object {
                        "label": 7,
                        "value": 7,
                      },
                      Object {
                        "label": 8,
                        "value": 8,
                      },
                      Object {
                        "label": 9,
                        "value": 9,
                      },
                      Object {
                        "label": 10,
                        "value": 10,
                      },
                      Object {
                        "label": 11,
                        "value": 11,
                      },
                      Object {
                        "label": 12,
                        "value": 12,
                      },
                      Object {
                        "label": 13,
                        "value": 13,
                      },
                      Object {
                        "label": 14,
                        "value": 14,
                      },
                      Object {
                        "label": 15,
                        "value": 15,
                      },
                      Object {
                        "label": 16,
                        "value": 16,
                      },
                      Object {
                        "label": 17,
                        "value": 17,
                      },
                      Object {
                        "label": 18,
                        "value": 18,
                      },
                      Object {
                        "label": 19,
                        "value": 19,
                      },
                      Object {
                        "label": 20,
                        "value": 20,
                      },
                      Object {
                        "label": 21,
                        "value": 21,
                      },
                      Object {
                        "label": 22,
                        "value": 22,
                      },
                      Object {
                        "label": 23,
                        "value": 23,
                      },
                      Object {
                        "label": 24,
                        "value": 24,
                      },
                      Object {
                        "label": 25,
                        "value": 25,
                      },
                      Object {
                        "label": 26,
                        "value": 26,
                      },
                      Object {
                        "label": 27,
                        "value": 27,
                      },
                      Object {
                        "label": 28,
                        "value": 28,
                      },
                      Object {
                        "label": 29,
                        "value": 29,
                      },
                      Object {
                        "label": 30,
                        "value": 30,
                      },
                      Object {
                        "label": 31,
                        "value": 31,
                      },
                      Object {
                        "label": 32,
                        "value": 32,
                      },
                      Object {
                        "label": 33,
                        "value": 33,
                      },
                      Object {
                        "label": 34,
                        "value": 34,
                      },
                      Object {
                        "label": 35,
                        "value": 35,
                      },
                      Object {
                        "label": 36,
                        "value": 36,
                      },
                      Object {
                        "label": 37,
                        "value": 37,
                      },
                      Object {
                        "label": 38,
                        "value": 38,
                      },
                      Object {
                        "label": 39,
                        "value": 39,
                      },
                      Object {
                        "label": 40,
                        "value": 40,
                      },
                      Object {
                        "label": 41,
                        "value": 41,
                      },
                      Object {
                        "label": 42,
                        "value": 42,
                      },
                      Object {
                        "label": 43,
                        "value": 43,
                      },
                      Object {
                        "label": 44,
                        "value": 44,
                      },
                      Object {
                        "label": 45,
                        "value": 45,
                      },
                      Object {
                        "label": 46,
                        "value": 46,
                      },
                      Object {
                        "label": 47,
                        "value": 47,
                      },
                      Object {
                        "label": 48,
                        "value": 48,
                      },
                      Object {
                        "label": 49,
                        "value": 49,
                      },
                      Object {
                        "label": 50,
                        "value": 50,
                      },
                      Object {
                        "label": 51,
                        "value": 51,
                      },
                      Object {
                        "label": 52,
                        "value": 52,
                      },
                      Object {
                        "label": 53,
                        "value": 53,
                      },
                      Object {
                        "label": 54,
                        "value": 54,
                      },
                      Object {
                        "label": 55,
                        "value": 55,
                      },
                      Object {
                        "label": 56,
                        "value": 56,
                      },
                      Object {
                        "label": 57,
                        "value": 57,
                      },
                      Object {
                        "label": 58,
                        "value": 58,
                      },
                      Object {
                        "label": 59,
                        "value": 59,
                      },
                      Object {
                        "label": 60,
                        "value": 60,
                      },
                      Object {
                        "label": 61,
                        "value": 61,
                      },
                      Object {
                        "label": 62,
                        "value": 62,
                      },
                      Object {
                        "label": 63,
                        "value": 63,
                      },
                      Object {
                        "label": 64,
                        "value": 64,
                      },
                      Object {
                        "label": 65,
                        "value": 65,
                      },
                      Object {
                        "label": 66,
                        "value": 66,
                      },
                      Object {
                        "label": 67,
                        "value": 67,
                      },
                      Object {
                        "label": 68,
                        "value": 68,
                      },
                      Object {
                        "label": 69,
                        "value": 69,
                      },
                      Object {
                        "label": 70,
                        "value": 70,
                      },
                      Object {
                        "label": 71,
                        "value": 71,
                      },
                      Object {
                        "label": 72,
                        "value": 72,
                      },
                      Object {
                        "label": 73,
                        "value": 73,
                      },
                      Object {
                        "label": 74,
                        "value": 74,
                      },
                      Object {
                        "label": 75,
                        "value": 75,
                      },
                      Object {
                        "label": 76,
                        "value": 76,
                      },
                      Object {
                        "label": 77,
                        "value": 77,
                      },
                      Object {
                        "label": 78,
                        "value": 78,
                      },
                      Object {
                        "label": 79,
                        "value": 79,
                      },
                      Object {
                        "label": 80,
                        "value": 80,
                      },
                      Object {
                        "label": 81,
                        "value": 81,
                      },
                      Object {
                        "label": 82,
                        "value": 82,
                      },
                      Object {
                        "label": 83,
                        "value": 83,
                      },
                      Object {
                        "label": 84,
                        "value": 84,
                      },
                      Object {
                        "label": 85,
                        "value": 85,
                      },
                      Object {
                        "label": 86,
                        "value": 86,
                      },
                      Object {
                        "label": 87,
                        "value": 87,
                      },
                      Object {
                        "label": 88,
                        "value": 88,
                      },
                      Object {
                        "label": 89,
                        "value": 89,
                      },
                      Object {
                        "label": 90,
                        "value": 90,
                      },
                      Object {
                        "label": 91,
                        "value": 91,
                      },
                      Object {
                        "label": 92,
                        "value": 92,
                      },
                      Object {
                        "label": 93,
                        "value": 93,
                      },
                      Object {
                        "label": 94,
                        "value": 94,
                      },
                      Object {
                        "label": 95,
                        "value": 95,
                      },
                      Object {
                        "label": 96,
                        "value": 96,
                      },
                      Object {
                        "label": 97,
                        "value": 97,
                      },
                      Object {
                        "label": 98,
                        "value": 98,
                      },
                    ]
                  }
                >
                  <div />
                </mockConstructor>
              </div>
            </div>
            <div
              className="collapsible mt-10"
            >
              <div
                className="collapsible-header"
              >
                <div
                  className="collapsible-title"
                >
                  <label
                    className="has-text-weight-bold"
                    htmlFor="other"
                  >
                    EQUINOX_STORAGE_SYSTEM
                  </label>
                </div>
              </div>
              <div
                className="field  mb-20 mt-20"
              >
                <mockConstructor
                  defaultValue=""
                  isSearchable={false}
                  onSelect={[Function]}
                  options={
                    Array [
                      Object {
                        "label": "NONE",
                        "value": "",
                      },
                      Object {
                        "label": "13KWH_1INV",
                        "value": "13kWh",
                      },
                    ]
                  }
                >
                  <div />
                </mockConstructor>
              </div>
            </div>
            <div
              className="collapsible mt-10"
            >
              <div
                className="collapsible-header"
              >
                <div
                  className="collapsible-title"
                >
                  <input
                    className="mr-10 checkbox"
                    defaultChecked={false}
                    id="other"
                    name="other"
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <label
                    className="has-text-weight-bold"
                    htmlFor="other"
                  >
                    OTHER_DEVICES
                  </label>
                </div>
              </div>
            </div>
          </div>
          <div>
            <p
              className="has-text-centered mt-20 mb-20"
            >
              RMA_ADD_DEVICES_TIP
            </p>
            <div
              className="rma-buttons"
            >
              <button
                className="button half-button-padding is-secondary is-uppercase trigger-scan mr-10"
                onClick={[Function]}
              >
                CANCEL
              </button>
              <button
                className="button half-button-padding is-primary is-uppercase trigger-scan"
                onClick={[Function]}
              >
                CONTINUE
              </button>
            </div>
          </div>
        </div>
      </RMAInventory>
    </Provider>
  </Router>
</BrowserRouter>
`;
